src/app/App.js

```diff
@@ -1,20 +1,50 @@
 import React, { Component } from "react";
 import { BrowserRouter, Route, Switch } from "react-router-dom";
 
+import Amplify from 'aws-amplify';
+import { withAuthenticator } from 'aws-amplify-react';
+import aws_exports from './aws-exports';
+
 import Map from "./map/Map";
-import Login from "./login/Login";
 import Form from "./form/Form";
 
 import "./App.css";
 
+Amplify.configure(aws_exports);
+
 class App extends Component {
+
+  componentWillMount(){
+    Amplify.Auth.currentAuthenticatedUser().then((user) => {
+      console.log('user',user);
+      user.getUserAttributes((err, attributes) => {
+        let o = {}; 
+        attributes.forEach(x => o[x.Name] = x.Value)
+        this.setState({user : o});
+      })
+    });
+  }
+
+  logout() {
+    Amplify.Auth.signOut()
+        .then(data => {
+          window.location.href=window.location.href;
```

Comment:
Are you reloading the page here? I think you could use `window.location.reload()` instead.

Reply:
Yes, they should be the same in the context. `window.location.reload()` is easier to read. I will replace it.